proto/regen/data/v1/tx.proto

```diff
@@ -112,11 +112,23 @@ message MsgDefineResolver {
   // the manager should make use of cosmos.authz.
   string manager = 1;
```

Comment:
Who would be the `manager` for the `ipfs:` resolver?

Can you elaborate on how you would expect this to work?  In reading the documentation for the `manager` field, it seems like only the `manager` of a resolver can register data to a given resolver.

So as I understand it, this PR implies that each user wanting to host data on IPFS would define their own resolver named `ipfs:` ?

I wonder if it would make more sense to instead extend the `MsgRegisterResolver()` message to have an enum field called `resolver_type`.  we could have the default / 0 value of that enum be `HTTP_RESOLVER`, and 1 = `IPFS_RESOLVER`.  This way, if we want to support other decentralized resolvers that are similar to IPFS in the future, we could just extend this enum?

Reply:
I think you're right. So, there's another step of work to do here. I don't think it will be too hard. I will close this, work on that and ping you when it's R4R again.

Reply:
@clevinson I've updated this PR to support marking a resolver as public to support IPFS. Please take a look again when you get a chance.